### PR TITLE
feat(elements): Form and Field registration/submission

### PR DESCRIPTION
**Pull request by tmilewski**
## Description

Enables Form and Field registration with state machine, along with associated React hooks.

Fixes inconsistencies in CI workflow caching of `node_modules`.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,11 +39,19 @@ jobs:
           turbo-team: ${{ vars.TURBO_TEAM }}
           turbo-token: ${{ secrets.TURBO_TOKEN }}
 
+      - name: List node_modules
+        run: npm ls
+        shell: bash
```
Comment (BRKalow) [inline .github/workflows/ci.yml:44]: Can we remove this?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -39,11 +39,19 @@ jobs:
...
+
       - name: Require Changeset
         timeout-minutes: ${{ fromJSON(vars.TIMEOUT_MINUTES_SHORT) }}
         if: ${{ !(github.event_name == 'merge_group') }}
         run: if [ "${{ github.event.pull_request.user.login }}" = "clerk-cookie" ]; then echo 'Skipping' && exit 0; else npx changeset status --since=origin/main; fi
 
+      - name: Lint GitHub Actions Workflows
+        run: npx eslint .github
+        shell: bash
```
Comment (BRKalow) [inline .github/workflows/ci.yml:53]: ❓ Why did we move this out of init?

Comment (tmilewski) [reply .github/workflows/ci.yml:53]: Shouldn't have stayed in init to begin with.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tmilewski) [reply .github/workflows/ci.yml:44]: We could though it errorred when the issues first occurred and could be a good litmus check.

Comment (BRKalow) [reply .github/workflows/ci.yml:44]: 👍 let's move it to init then, wdyt?

**Review by BRKalow — APPROVED** [review]
lgtm 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tmilewski) [reply .github/workflows/ci.yml:44]: I'll do it in a follow-up

**Pull request merged**